crates/solver/src/settlement_submission/submitter.rs

```diff
@@ -13,91 +13,109 @@
 // from outside) so it is only at that point that we need to check the hashes individually to the
 // find the one that got mined (if any).
 
-use super::{flashbots_api::FlashbotsApi, ESTIMATE_GAS_LIMIT_FACTOR};
-use crate::settlement::Settlement;
+use super::ESTIMATE_GAS_LIMIT_FACTOR;
+use crate::{interactions::block_coinbase, settlement::Settlement};
 use anyhow::{anyhow, ensure, Context, Error, Result};
 use contracts::GPv2Settlement;
-use ethcontract::{transaction::Transaction, Account};
+use ethcontract::{contract::MethodBuilder, dyns::DynTransport, transaction::Transaction, Account};
 use futures::FutureExt;
 use gas_estimation::{EstimatedGasPrice, GasPriceEstimating};
 use primitive_types::{H256, U256};
 use shared::Web3;
 use std::time::{Duration, Instant, SystemTime};
 use web3::types::TransactionReceipt;
 
-pub struct FlashbotsSolutionSubmitter<'a> {
+/// Parameters for transaction submitting
+#[derive(Clone, Default)]
+pub struct SubmitterParams {
+    /// Desired duration to include the transaction in a block
+    pub target_confirm_time: Duration, //todo change to blocks in the following PR
+    /// Estimated gas consumption of a transaction
+    pub gas_estimate: U256,
+    /// Maximum max_fee_per_gas to pay for a transaction
+    pub gas_price_cap: f64,
+    /// Maximum duration of a single run loop
+    pub deadline: Option<SystemTime>,
+    /// There are two ways to pay for a transaction:
+    /// 1. Set valid gas price (default)
+    /// 2. Set 0 gas price, but add additional step to settlement to pay directly to the coinbase
+    /// Contains quantity of gas needed to pay for the additional step
+    pub pay_gas_to_coinbase: Option<U256>,
+    /// Boost estimated gas price miner tip in order to increase the chances of a transaction being mined
+    pub additional_miner_tip: Option<f64>,
+    /// Resimulate and resend transaction on every retry_interval seconds
+    pub retry_interval: Duration,
+}
+
+#[async_trait::async_trait]
+pub trait TransactionSubmitting {
+    async fn submit_raw_transaction(&self, tx: &[u8], params: &SubmitterParams) -> Result<String>;
+    async fn cancel_transaction(&self, id: &str) -> Result<()>;
```

Comment:
Would like comments here to explain the semantics of these functions. If I was looking at this struct and trait in isolation without having seen the previous implementation I feel I would not know what to do here.

Reply:
Added.